source/incqueryserver-jupyter/iqs_jupyter/helpers.py

```diff
@@ -31,16 +31,30 @@ def cell_to_html(cell):
     else:
         return str(cell)
 
-def dict_to_element(dict_or_attribute_value, url_provider=None):
+def dict_to_element(dict_or_attribute_value):
     if isinstance(dict_or_attribute_value, dict):
         for recognizer in ext_point.element_dict_recognizers:
             recognized = recognizer(dict_or_attribute_value)
             if recognized:
-                if url_provider is not None:
-                    recognized.url = url_provider(recognized)
+                for url_provider in ext_point.url_providers:
+                    if url_provider(dict_or_attribute_value['element']) is not None: # maybe we should set url param to None - else
+                        setattr(recognized, 'url', url_provider(dict_or_attribute_value['element']))
```

Comment:
Thank you for your comments!

0.  There is no `url` field of elements in the case of query execution
1.  Later `url` field can be a list, or a "domain" to "url" map in order to support multiple urls for an element.
     At this point it is important to note that in IQS 2021.5.0 (coming in December) will be an API endpoint for providing 
     links to original service, IncQuery Desktop and Model Viewer.
2. We will have to enhance the html representation to handle multiple links for an element

Reply:
0. In Python, you can just add fields to most objects by simple assignment - is there a reason why it would not work in this case? The last time I tested the original url_provider solution it still worked - but that was long ago, is there a reason the assignment needed to be replaced? Did the generated Python client change in a way that now the protocol data objects are no longer dict-backed?
1. Yes, I agree; I think dict would be the especially best. Anyway, either have a single url provider logic hardcoded for the time being, or have it as an extensible list, but then properly handle the list having multiple elements.
2. Yes, but that HTML representation is generated here, so it would make sense to have it as part of this PR.